model/riscv_insts_vext_mem.sail

```diff
@@ -154,6 +154,7 @@ function process_vlsegff (nf, vm, vd, load_width_bytes, rs1, EMUL_pow, num_elem)
               if i == 0 then { ext_handle_data_check_error(e); return RETIRE_FAIL }
               else {
                 vl = to_bits(sizeof(xlen), i);
+                if rv_enable_callbacks then csr_update_callback("vl", vl);
```

Comment:
Having to add these manually is rather error prone and makes the code less readable. Can we do some fancy sail overloading tricks to automatically call these functions for CSR writes (e.g. something like `CSR(vl) = ...` ?

Reply:
Various CSRs have a `set_foo()` function. We could put it in there. Though that doesn't significantly reduce the risk because people can always forget to use the setter.

I ran into a similar issue recently - trying to memoize/cache a value. Difficult to guarantee that all the code will use setter that updates the cache without some support for `private`.

IMO there are few enough of these that we can probably ignore this issue for the moment, though I agree it's not ideal.

Reply:
Perhaps it would be better to get Sail to generate the callbacks automatically?

Reply:
Yeah we discussed this before. Might be nice, but I'm not 100% sure it would do exactly what you want, e.g. you probably want distinct callbacks for writes to `sstatus` and `mstatus`, but under the hood they write to the same `register`. Also there are very few places this callback is needed so the manual burden is not that high, and I suspect Alasdair has enough on his plate already!

Reply:
I added the `ext_notification_write_CSR` (and `ext_notification_read_CSR`) function. It helps to simplify writing to a CSR with a call to a callback.